sklearn/utils/openmp_helpers.pyx

```diff
@@ -0,0 +1,37 @@
+IF SKLEARN_OPENMP_SUPPORTED:
+    cimport openmp
+    from joblib import cpu_count
+
+
+cpdef _openmp_effective_n_threads(n_threads=None):
+    """Determine the effective number of threads used for parallel OpenMP calls
+
+    - For ``n_threads = None``, returns the minimum between
+      openmp.omp_get_max_threads() and joblib.effective_n_jobs(-1).
+      The result of ``omp_get_max_threads`` can be influenced by environment
+      variable ``OMP_NUM_THREADS`` or at runtime by ``omp_set_num_threads``.
+    - For ``n_threads > 0``, use this as the maximal number of threads for
+      parallel OpenMP calls.
+    - For ``n_threads < 0``, use the maximal number of threads minus
+      ``|n_threads + 1|``.
```

Comment:
Maybe expand: "In particular `n_threads=-1` will use as many threads as there are available cores on the machine.".